src/common.c

```diff
@@ -2951,19 +2951,13 @@ int stlink_erase_flash_page(stlink_t *sl, stm32_addr_t flashaddr) {
   return check_flash_error(sl);
 }
 
-int stlink_erase_flash_mass(stlink_t *sl) {
-  int err = 0;
-
-  // TODO: User MER bit to mass-erase WB series.
-  if (sl->flash_type == STLINK_FLASH_TYPE_L0 ||
-      sl->flash_type == STLINK_FLASH_TYPE_WB) {
+int stlink_erase_flash_section(stlink_t *sl, stm32_addr_t base_addr, size_t size) {
     // erase each page
-    int i = 0, num_pages = (int)(sl->flash_size / sl->flash_pgsz);
-
+    int i = 0, num_pages = (int)(size / sl->flash_pgsz);
```

Comment:
You cannot calculate the number of memory segments by simple division. In reality, many MCUs have sectors of different sizes.

Reply:
Addressed by https://github.com/stlink-org/stlink/pull/1213#issuecomment-1008361996 and the following discussion.